windows/hid.c

```diff
@@ -238,19 +245,31 @@ static void register_error(hid_device *dev, const char *op)
 
 	/* Get rid of the CR and LF that FormatMessage() sticks at the
 	   end of the message. Thanks Microsoft! */
-	ptr = msg;
-	while (*ptr) {
-		if (*ptr == L'\r') {
-			*ptr = L'\0';
-			break;
-		}
-		ptr++;
+	msg[msg_size - 2] = L'\0';
+	msg_size -= 2;
```

Comment:
Does the cocumentation of `FormatMessageW` guarantees that the message will end with `\r\n`?

I don't like this approach (nor the previous one).

Technically correct would be:
```
while(msg_size > 0 && (msg[msg_size-1] == '\r' || msg[msg_size-1] == '\n'))
	msg_size--;
```